From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Neurai <dev@neurai.org>
Date: Mon, 14 Apr 2026 00:00:00 +0000
Subject: [PATCH] Fix missing #include <limits> in qglobal.h for GCC 12+

GCC 12 no longer transitively includes <limits> through other standard
headers, causing std::numeric_limits to be undeclared in qendian.h,
qbytearraymatcher.h and other Qt headers. Adding it to qglobal.h fixes
all of them at once since every Qt header includes qglobal.h.
---
 qtbase/src/corelib/global/qglobal.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/qtbase/src/corelib/global/qglobal.h b/qtbase/src/corelib/global/qglobal.h
--- a/qtbase/src/corelib/global/qglobal.h
+++ b/qtbase/src/corelib/global/qglobal.h
@@ -46,6 +46,7 @@
 #ifdef __cplusplus
 #  include <type_traits>
 #  include <cstddef>
+#  include <limits>
 #  include <utility>
 #endif
 #ifndef __ASSEMBLER__

